subsys/bluetooth/host/classic/sdp.c

```diff
@@ -1086,7 +1092,7 @@ static uint16_t sdp_svc_att_req(struct bt_sdp *sdp, struct net_buf *buf,
 	max_att_len = net_buf_pull_be16(buf);
 
 	/* Set up the filters */
-	res = get_att_search_list(buf, filter, &num_filters);
+	res = get_att_search_list(buf, filter, MAX_NUM_ATT_ID_FILTER, &num_filters);
```

Comment:
I'd use `ARRAY_SIZE(filter)` instead of the macro. That way this code is always correct even if you change the array definition.

Reply:
Updated